datastore/query_test.go

```diff
@@ -500,6 +500,77 @@ var (
 	}
 )
 
+func TestPropertyFilterToProto(t *testing.T) {
+	testCases := []struct {
+		pf         PropertyFilter
+		wantErrMsg string
+	}{
+		{PropertyFilter{"x", "=", 4}, ""},
+		{PropertyFilter{"", "=", 4}, "datastore: empty query filter field name"},
+		{PropertyFilter{"x", "==", 4}, "datastore: unknown query filter operator"},
+		{PropertyFilter{"x", "==", struct{ x string }{x: "sample"}}, "datastore: bad query filter value type: invalid Value type struct { x string }"},
+	}
+
+	for _, tc := range testCases {
+		_, err := tc.pf.toProto()
+		gotErrMsg := ""
+		if err != nil {
+			gotErrMsg = err.Error()
+		}
+
+		if gotErrMsg != tc.wantErrMsg {
+			t.Errorf("PropertyFilter proto conversion error: \nwant %v,\ngot %v", tc.wantErrMsg, gotErrMsg)
+		}
+	}
+}
+
+func TestCompositeFilterToProto(t *testing.T) {
+	testCases := []struct {
+		cf         CompositeFilter
+		wantErrMsg string
+	}{
+		{&AND{[]EntityFilter{&PropertyFilter{"x", "=", 4}}}, ""},
```

Comment:
Same as above--consider whether there are other valid EntityFilters that you should have testCases for.

Reply:
Done. Added all the testcases that were written for FilterField